File: sguyms.py
import os
import json

variables = {}

class Variable:
	def __init__(self, name, value):
		self.value = value
		self.name = name
	def hold(self):
		global variables
		variables[self.name] = self.value

def get(variable):
	return variables[variable]

def call_py(script_id, func_name, cmd_start = False, args = ''):
	with open(f'./running/call.py', 'w') as f:
		f.write(f'from running{script_id} import *\n')
		f.write(f'{func_name}({args})')
	if cmd_start:
		os.system('start python ./running/call.py')
	os.system('python ./running/call.py')

def call_node(script_id, func_name, cmd_start = False, args = ''):
	with open(f'./running/call.js', 'w') as f:
		f.write(f'let sgms = require("./running{script_id}.js");\n')
		f.write(f'sgms.{func_name}({args});')
	if cmd_start:
		os.system('start node ./running/call.js')
	os.system('node ./running/call.js')

def load():
	global variables
	with open('./running/variables.json', 'r', encoding='utf-8') as f:
		variables = json.load(f)

def save():
	with open('./running/variables.json', 'w', encoding='utf-8') as f:
		json.dump(variables, f, ensure_ascii=False, indent=4)

File: sguyms.js
const { exec } = require('child_process');
const fs = require('fs');

let variables = {}

class Variable{
    constructor(name, value) {
        this.name = name;
        this.value = value;
    }

    hold(){
        variables[this.name] = this.value;
    }
}

function get(variable) {
    return variables[variable]
}

function call_py(script_id, func_name, args = ''){
    fs.writeFile('./running/call.py', `from running${script_id} import *\n${func_name}(${args})`, 'utf8', (writeErr) => {
        if (writeErr) {
            console.error(writeErr);
            return;
        }

        exec('python ./running/call.py', (execError, stdout, stderr) => {
            if (execError) {
                console.error(execError);
                return;
            }
            if (stderr) {
                console.error(stderr);
                return;
            }
            console.log(stdout);
        });
    });
}

function call_node(script_id, func_name, args = ''){
    fs.writeFile('./running/call.js', `let sgms = require("./running${script_id}.js");\nsgms.${func_name}(${args});`, 'utf8', (writeErr) => {
        if (writeErr) {
            console.error(writeErr);
            return;
        }

        exec('node ./running/call.js', (execError, stdout, stderr) => {
            if (execError) {
                console.error(execError);
                return;
            }
            if (stderr) {
                console.error(stderr);
                return;
            }
            console.log(stdout);
        });
    });
}

function load(){
    try {
      const data = fs.readFileSync('./running/variables.json', 'utf8');
      try {
          const variables = JSON.parse(data);
      } catch (error) {
          console.error(error.message);
      }
    } catch (err) {
      console.error(err);
    }
}

function save(){
    try {
      fs.writeFileSync('./running/variables.json', JSON.stringify(variables), 'utf8');
    } catch (err) {
      console.error(err);
    }
}

module.exports = {
    call_py,
    load,
    save,
    get,
    call_node,
    Variable
}